lib/pos_epoch.go

```diff
@@ -0,0 +1,170 @@
+package lib
+
+import (
+	"bytes"
+	"github.com/dgraph-io/badger/v3"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+)
+
+//
+// TYPE
+//
+
+type EpochEntry struct {
+	EpochNumber      uint64
+	FinalBlockHeight uint64
+}
+
+func (epochEntry *EpochEntry) Copy() *EpochEntry {
+	return &EpochEntry{
+		EpochNumber:      epochEntry.EpochNumber,
+		FinalBlockHeight: epochEntry.FinalBlockHeight,
+	}
+}
+
+func (epochEntry *EpochEntry) RawEncodeWithoutMetadata(blockHeight uint64, skipMetadata ...bool) []byte {
+	var data []byte
+	data = append(data, UintToBuf(epochEntry.EpochNumber)...)
+	data = append(data, UintToBuf(epochEntry.FinalBlockHeight)...)
+	return data
+}
+
+func (epochEntry *EpochEntry) RawDecodeWithoutMetadata(blockHeight uint64, rr *bytes.Reader) error {
+	var err error
+
+	// EpochNumber
+	epochEntry.EpochNumber, err = ReadUvarint(rr)
+	if err != nil {
+		return errors.Wrapf(err, "EpochEntry.Decode: Problem reading EpochNumber: ")
+	}
+
+	// FinalBlockHeight
+	epochEntry.FinalBlockHeight, err = ReadUvarint(rr)
+	if err != nil {
+		return errors.Wrapf(err, "EpochEntry.Decode: Problem reading FinalBlockHeight: ")
+	}
+
+	return err
+}
+
+func (epochEntry *EpochEntry) GetVersionByte(blockHeight uint64) byte {
+	return 0
+}
+
+func (epochEntry *EpochEntry) GetEncoderType() EncoderType {
+	return EncoderTypeEpochEntry
+}
+
+//
+// UTXO VIEW UTILS
+//
+
+func (bav *UtxoView) GetCurrentEpochEntry() (*EpochEntry, error) {
+	var epochEntry *EpochEntry
+	var err error
+
+	// First, check the UtxoView.
+	epochEntry = bav.CurrentEpochEntry
+	if epochEntry != nil {
+		return epochEntry.Copy(), nil
```

Comment:
I think normally we don't Copy() before returning from the UtxoView Getters. I don't think it matters, but probably good to keep the same convention as everywhere else.